### PR TITLE
Add Gensim Support

### PR DESCRIPTION
**Pull request by koaning**
Gensim provides one of the easier methods of training your own word embeddings. It was suggested to me that this might be a best practice for some African languages because internet language there does not resemble their wikipedia pages at all.

### CODE REVIEW

**Comment by koaning** [comment]
Note that this feature should resolve https://github.com/RasaHQ/rasa-nlu-examples/issues/12.

File: rasa_nlu_examples/featurizers/dense/gensim_featurizer.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        features = self._combine_with_existing_dense_features(
+            message, additional_features=X, feature_name=DENSE_FEATURE_NAMES[attribute]
+        )
+        message.set(DENSE_FEATURE_NAMES[attribute], features)
+
+    def process(self, message: Message, **kwargs: Any) -> None:
+        self.set_gensim_features(message)
+
+    def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
+        pass
```
Comment (tabergma) [inline rasa_nlu_examples/featurizers/dense/gensim_featurizer.py:86]:
```suggestion
        return None
```

Comment (tabergma) [reply rasa_nlu_examples/featurizers/dense/gensim_featurizer.py:86]: Not sure if it is needed.

File: rasa_nlu_examples/featurizers/dense/gensim_featurizer.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        tokens = message.get(TOKENS_NAMES[attribute])
+
+        if not tokens:
+            return None
+
+        word_vectors = [
+            self.kv[t.text] if t.text in self.kv else np.zeros(self.kv.vector_size)
+            for t in train_utils.tokens_without_cls(message, attribute)
+        ]
+        text_vector = reduce(lambda a, b: a + b, word_vectors)
```
Comment (tabergma) [inline rasa_nlu_examples/featurizers/dense/gensim_featurizer.py:74]: Would maybe add a comment. Something like `Sum up all word vectors as text vector`.

**Review by tabergma — APPROVED** [review]
Looks great! 🚀  Added a few comments.

File: docs/index.md [hunk 3]
```diff
@@ -26,6 +26,13 @@ You can find a tutorial on it [here](https://blog.rasa.com/custom-printer-compon
 
 ### Dense Featurizers
 
+#### GensimFeaturizer
+
+**`rasa_nlu_examples.featurizers.dense.GensimFeaturizer` [docs](docs/featurizer/gensim/)**
+
+A benefit of `gensim` is that it is very easy to train your own word embeddings. This
+featurizer allows you to train your own and then to use it in a Rasa pipeline.
```
[the remark this answers is not in the document]
Comment (koaning) [reply docs/index.md:34]: I'll rephrase. The thing with `gensim` is that it doesn't really host pretrained models that much but it only needs about 4 lines of code to train your own. I'd argue training your own is the main reason I'm interested in it. There were some folks in Afrika asking about this.

File: rasa_nlu_examples/featurizers/dense/gensim_featurizer.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    @classmethod
+    def required_packages(cls) -> List[Text]:
+        return ["gensim"]
+
+    defaults = {"file": None, "cache_dir": None}
+    language_list = None
+
+    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
+        super().__init__(component_config)
+        path = os.path.join(component_config["cache_dir"], component_config["file"])
```
[the remark this answers is not in the document]
Comment (koaning) [reply rasa_nlu_examples/featurizers/dense/gensim_featurizer.py:41]: They should be both set at all times. Ah, but maybe I need to specify that.

File: docs/docs/featurizer/gensim.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+file that you can pass on to this component.
+
+## Configurable Variables
+
+- **cache_dir**: pass it the name of the directory where you've downloaded/saved the embeddings
+- **file**: pass it the name of the `.kv` file that contains the word embeddings
+
+## Base Usage
+
+The configuration file below demonstrates how you might use the fasttext embeddings. In this example
```
[the remark this answers is not in the document]
Comment (koaning) [reply docs/docs/featurizer/gensim.md:32]: 😅

File: rasa_nlu_examples/featurizers/dense/fasttext_featurizer.py [hunk 6]
```diff
@@ -37,6 +37,10 @@ def required_packages(cls) -> List[Text]:
 
     def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
         super().__init__(component_config)
+        if not component_config["cache_dir"]:
+            raise ValueError("You need to set `cache_dir` for the GensimFeaturizer.")
```
Comment (tabergma) [inline rasa_nlu_examples/featurizers/dense/fasttext_featurizer.py:41]:
```suggestion
            raise ValueError("You need to set `cache_dir` for the FasttextFeaturizer.")
```

File: rasa_nlu_examples/featurizers/dense/fasttext_featurizer.py [hunk 7]
```diff
@@ -37,6 +37,10 @@ def required_packages(cls) -> List[Text]:
...
+        if not component_config["file"]:
+            raise ValueError("You need to set `file` for the GensimFeaturizer.")
```
Comment (tabergma) [inline rasa_nlu_examples/featurizers/dense/fasttext_featurizer.py:43]:
```suggestion
            raise ValueError("You need to set `file` for the FasttextFeaturizer.")
```

**Pull request merged**